src/main/java/de/tum/p2p/api/netty/forwarding/OnionTunnelDestroyMessage.java

```diff
@@ -0,0 +1,35 @@
+package de.tum.p2p.api.netty.forwarding;
+
+import de.tum.p2p.onion.forwarding.TunnelId;
+import de.tum.p2p.proto.message.MessageType;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.nio.ByteBuffer;
+
+@ToString
+@EqualsAndHashCode(callSuper = true)
+public class OnionTunnelDestroyMessage extends OnionApiMessage {
+
+    private final TunnelId tunnelId;
+
+    public OnionTunnelDestroyMessage(TunnelId id) {
+        super(MessageType.ONION_TUNNEL_DESTROY);
+        this.tunnelId = id;
+    }
+
+    public TunnelId getTunnelId() {
+        return tunnelId;
+    }
+
+    @Override
+    protected ByteBuffer writeMessage(ByteBuffer typedMessageBuffer) {
+        return null;
+    }
+
+    public static OnionTunnelDestroyMessage fromBytes(ByteBuffer buffer) throws Exception {
+        int id = buffer.getInt();
```

Comment:
This one will be not int. First Short in the ByteBuffer will be MessageType (short).